apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1916,10 +1895,12 @@ where
             }
         }
         if let ConditionResolution::Satisfied {
+            cost,
             context_map: ctx_map,
             ..
         } = &mut resolution
         {
+            *cost += total_cost;
```

Comment:
Bugs:
- The cost of context conditions was being ignored when there were also edge conditions.